### PR TITLE
isis device tests

### PR DESCRIPTION
**Pull request by ANISH-GOTTAPU**

### CODE REVIEW

File: features/devices/isis/isis_route_install_test.go [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+		"txAdvRouteV4": "10.10.10.1",
+		"rxAdvRouteV4": "20.20.20.1",
+		"txAdvRouteV6": "::10:10:10:01",
+		"rxAdvRouteV6": "::20:20:20:01",
+	}
+
+	api := otg.NewOtgApi(t)
+	c := isisRouteInstallConfig(api, testConst)
+
+	fmt.Println(c.ToJson())
```
Comment (ashutshkumr) [inline features/devices/isis/isis_route_install_test.go:45]: remove this

File: features/devices/isis/isis_route_install_test.go [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+	c.Layer1().Add().
+		SetName("ly").
+		SetPortNames([]string{ptx.Name(), prx.Name()}).
+		SetSpeed(gosnappi.Layer1SpeedEnum(api.TestConfig().OtgSpeed))
+
+	tx := c.Devices().Add().SetName("tx")
+	rx := c.Devices().Add().SetName("rx")
+
+	// transmit
+	txEth := tx.Ethernets().
```
Comment (ashutshkumr) [inline features/devices/isis/isis_route_install_test.go:79]: add "d" as a prefix for all device objects (similar to what we do for bgp)

File: features/devices/isis/isis_route_install_test.go [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+		Ipv6Addresses().
+		Add().
+		SetName("txIpv6").
+		SetAddress(tc["txIpv6"].(string)).
+		SetGateway(tc["txv6Gateway"].(string)).
+		SetPrefix(tc["txv6Prefix"].(int32))
+
+	txIsis := tx.Isis().
+		SetSystemId("640000000001").
+		SetName("tx_isis")
```
Comment (ashutshkumr) [inline features/devices/isis/isis_route_install_test.go:104]: follow the camelCase convention for naming objects

File: features/devices/isis/isis_route_install_test.go [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+
+	txIsis.Basic().
+		SetIpv4TeRouterId(tc["txIp"].(string)).
+		SetHostname("ixia-c-port1")
```
Comment (ashutshkumr) [inline features/devices/isis/isis_route_install_test.go:108]: Use object name as hostname ? Would suggest not hardcoding this.

File: features/devices/isis/isis_route_install_test.go [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+	txEth.
+		Ipv6Addresses().
+		Add().
+		SetName("txIpv6").
+		SetAddress(tc["txIpv6"].(string)).
+		SetGateway(tc["txv6Gateway"].(string)).
+		SetPrefix(tc["txv6Prefix"].(int32))
+
+	txIsis := tx.Isis().
+		SetSystemId("640000000001").
```
Comment (ashutshkumr) [inline features/devices/isis/isis_route_install_test.go:103]: do not hardcode this

File: features/devices/isis/isis_route_install_test.go [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+		SetName("tx_isis")
+
+	txIsis.Basic().
+		SetIpv4TeRouterId(tc["txIp"].(string)).
+		SetHostname("ixia-c-port1")
+
+	txIsis.Advanced().
+		SetAreaAddresses([]string{"490001"}).
```
Comment (ashutshkumr) [inline features/devices/isis/isis_route_install_test.go:111]: do not hardcode this

File: features/devices/isis/isis_route_install_test.go [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+	txEth.
+		Ipv4Addresses().
+		Add().
+		SetName("txIp").
+		SetAddress(tc["txIp"].(string)).
+		SetGateway(tc["txGateway"].(string)).
+		SetPrefix(tc["txPrefix"].(int32))
+
+	txEth.
+		Ipv6Addresses().
```
Comment (ashutshkumr) [inline features/devices/isis/isis_route_install_test.go:95]: When exactly are we using v4 and v6 ip addresses ? what happens if we get rid of these ?

File: features/devices/isis/isis_route_install_test.go [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+	frxV6Tcp.SrcPort().SetValue(6000)
+	frxV6Tcp.DstPort().SetValue(5000)
+
+	api.Testing().Logf("Config:\n%v\n", c)
+	return c
+}
+
+func isisRouteInstallMetricsOk(api *otg.OtgApi, tc map[string]interface{}) bool {
+	for _, m := range api.GetIsIsMetrics() {
+		if m.L1SessionsUp() != 1 || m.L2SessionsUp() != 1 ||
```
Comment (ashutshkumr) [inline features/devices/isis/isis_route_install_test.go:296]: can we not hardcode these values ?

why are we checking L1/L2 database size btw ? Please note, this check does not ensure that routes are correctly advertised. Hence, for that, one MUST rely on isis states (not supported as of now).

so would suggest adding comments accordingly.

File: features/devices/isis/test_isis_route_install.py [hunk 9]
```diff
@@ -0,0 +1,244 @@
+import logging as log
+import pytest
+from helpers.otg import otg
+
```
Comment (ashutshkumr) [inline features/devices/isis/test_isis_route_install.py:4]: comments similar to go test applies here as well

File: helpers/otg/metrics.go [hunk 10]
```diff
@@ -88,3 +88,42 @@ func (o *OtgApi) GetBgpv4Metrics() []gosnappi.Bgpv4Metric {
...
+	tb := table.NewTable(
+		"ISIS Metrics",
+		[]string{
+			"Name",
+			"L1 Sessions Up",
+			"L2 Sessions UP",
+			"L1 Database Size",
+			"L2 Database Size",
+		},
+		15,
```
Comment (ashutshkumr) [inline helpers/otg/metrics.go:113]: add more spacing in columns - current one doesn't look ok

File: helpers/otg/otg.py [hunk 11]
```diff
@@ -264,6 +264,42 @@ def get_bgpv4_metrics(self):
...
+
+            tb = table.Table(
+                "ISIS Metrics",
+                [
+                    "Name",
+                    "L1 Sessions Up",
+                    "L2 Sessions UP",
+                    "L1 Database Size",
+                    "L2 Database Size",
+                ],
```
Comment (ashutshkumr) [inline helpers/otg/otg.py:284]: add more spacing in columns - current one doesn't look ok

**Review by ashutshkumr — CHANGES_REQUESTED** [review]
Please sync with main.

Also, the test should be called `isis_p2p_l12_route_install`

File: features/devices/isis/isis_route_install_test.go (hunk 7 above)
Comment (ANISH-GOTTAPU) [reply features/devices/isis/isis_route_install_test.go:97]: If we get rid of v4 or v6 address, we are getting error from the server

ERROR:  {
          "errors":  [
            "Error occurred while starting flows: [error starting tx port ptx: unsuccessful Response: BAD_REQUEST]"
          ]
        }

File: features/devices/isis/isis_route_install_test.go (hunk 8 above)
Comment (ANISH-GOTTAPU) [reply features/devices/isis/isis_route_install_test.go:300]: Added TODO Comment

File: features/devices/isis/isis_route_install_test.go (hunk 1 above)
Comment (ANISH-GOTTAPU) [reply features/devices/isis/isis_route_install_test.go]: updated

File: features/devices/isis/isis_route_install_test.go (hunk 2 above)
Comment (ANISH-GOTTAPU) [reply features/devices/isis/isis_route_install_test.go]: updated

File: features/devices/isis/isis_route_install_test.go (hunk 3 above)
Comment (ANISH-GOTTAPU) [reply features/devices/isis/isis_route_install_test.go]: updated

File: features/devices/isis/isis_route_install_test.go (hunk 4 above)
Comment (ANISH-GOTTAPU) [reply features/devices/isis/isis_route_install_test.go]: updated

File: features/devices/isis/test_isis_route_install.py (hunk 9 above)
Comment (ANISH-GOTTAPU) [reply features/devices/isis/test_isis_route_install.py:4]: updated

File: helpers/otg/otg.py (hunk 11 above)
Comment (ANISH-GOTTAPU) [reply helpers/otg/otg.py:284]: updated

**Review by ashutshkumr — APPROVED** [review]

**Pull request merged**